### PR TITLE
FIX: removed the duplicated staged users without posts

### PR DESCRIPTION
**Pull request by LeoMcA**
@tgxworld this is changing some of your code, you should probably take a look

### CODE REVIEW

File: app/jobs/onceoff/fix_primary_emails_for_staged_users.rb [hunk 1]
```diff
@@ -10,11 +10,10 @@ def execute_onceoff(args)
         .each_key do |email|
 
         users.where("email_tokens.email = ?", email)
-          .order(id: :asc)
-          .offset(1)
+          .order(id: :desc)
```
Comment (tgxworld) [inline app/jobs/onceoff/fix_primary_emails_for_staged_users.rb]: Why is the ordering changed? Wouldn't this destroy all the staged users?

File: app/jobs/onceoff/fix_primary_emails_for_staged_users.rb [hunk 2]
```diff
@@ -10,11 +10,10 @@ def execute_onceoff(args)
...
           .each do |user|
 
-          destroyer.destroy(user)
+          destroyer.destroy(user) unless user.posts.count > 0 || users.where("email_tokens.email = ?", email).count == 1
```
Comment (tgxworld) [inline app/jobs/onceoff/fix_primary_emails_for_staged_users.rb]: Could you elaborate on why this condition is needed here?

File: spec/jobs/fix_primary_emails_for_staged_users_spec.rb [hunk 3]
```diff
@@ -1,25 +1,35 @@
...
 
+  it 'should clean up duplicated staged users' do
     expect { described_class.new.execute_onceoff({}) }
       .to change { User.count }.by(-2)
 
     expect(User.all).to contain_exactly(Discourse.system_user, staged_user, active_user)
     expect(staged_user.reload.email).to eq(common_email)
   end
+
+  it 'should keep the duplicated staged user with posts' do
```
Comment (tgxworld) [inline spec/jobs/fix_primary_emails_for_staged_users_spec.rb]: Ah OK I see what is happening. So both the original and duplicated staged users have posted? If that is the case, we're need to transfer all the posts to the original staged user and nuke the duplicated user.

**Comment by LeoMcA** [comment]
@tgxworld revised and force pushed (to update my commit comment)

File: app/jobs/onceoff/fix_primary_emails_for_staged_users.rb [hunk 4]
```diff
@@ -2,18 +2,25 @@ module Jobs
...
       users.group("email_tokens.email")
         .having("COUNT(email_tokens.email) > 1")
         .count
         .each_key do |email|
 
-        users.where("email_tokens.email = ?", email)
-          .order(id: :asc)
-          .offset(1)
+        query = users.where("email_tokens.email = ?", email)
+                  .order(id: :asc)
```
Comment (tgxworld) [inline app/jobs/onceoff/fix_primary_emails_for_staged_users.rb]: Indentation looks abit off here :grin: If it fits within 80 chars, we can just put it on the same line

File: app/jobs/onceoff/fix_primary_emails_for_staged_users.rb [hunk 5]
```diff
@@ -2,18 +2,25 @@ module Jobs
...
+
+        original_user = query.first
+
+        query.offset(1)
           .each do |user|
```
Comment (tgxworld) [inline app/jobs/onceoff/fix_primary_emails_for_staged_users.rb]: let's jut move this to the same line as the code above :+1:

File: spec/jobs/fix_primary_emails_for_staged_users_spec.rb [hunk 6]
```diff
@@ -1,25 +1,36 @@
...
     expect(staged_user.reload.email).to eq(common_email)
   end
+
+  it 'should move posts owned by duplicate users to the original' do
+    post1 = Fabricate(:post, user: staged_user2)
+    post2 = Fabricate(:post, user: staged_user2)
+    post3 = Fabricate(:post, user: staged_user3)
+
+    expect { described_class.new.execute_onceoff({}) }
+      .to change { staged_user.posts.count }.by(+3)
```
Comment (tgxworld) [inline spec/jobs/fix_primary_emails_for_staged_users_spec.rb]: Just `3` will be sufficient.

File: spec/jobs/fix_primary_emails_for_staged_users_spec.rb [hunk 7]
```diff
@@ -1,25 +1,36 @@
...
 
+  it 'should clean up duplicated staged users' do
     expect { described_class.new.execute_onceoff({}) }
       .to change { User.count }.by(-2)
 
     expect(User.all).to contain_exactly(Discourse.system_user, staged_user, active_user)
     expect(staged_user.reload.email).to eq(common_email)
   end
+
+  it 'should move posts owned by duplicate users to the original' do
```
Comment (tgxworld) [inline spec/jobs/fix_primary_emails_for_staged_users_spec.rb]: I would actually move the assertions of this test together with `should clean up duplicated staged users`

**Comment by tgxworld** [comment]
Thanks for the fixes @LeoMcA Looks great to me.

**Comment by LeoMcA** [comment]
@tgxworld okay, reformatted the things you suggested (and remembered to run rubocop this time 😉), and reverted the test to how it was originally, including the post assertions

**Comment by tgxworld** [comment]
Looking good to me @LeoMcA :+1: Thanks for working on this.

**Pull request merged**